extendeddatasquare_test.go

```diff
@@ -59,119 +59,88 @@ func TestComputeExtendedDataSquare(t *testing.T) {
 
-	var eds ExtendedDataSquare
-	err = json.Unmarshal(edsBytes, &eds)
-	if err != nil {
-		t.Errorf("failed to marshal EDS: %v", err)
-	}
-	if !reflect.DeepEqual(result.squareRow, eds.squareRow) {
-		t.Errorf("eds not equal after json marshal/unmarshal")
-	}
+	var got ExtendedDataSquare
+	err = json.Unmarshal(edsBytes, &got)
+	require.NoError(t, err)
+
+	assert.Equal(t, original.dataSquare.Flattened(), got.dataSquare.Flattened())
+	assert.Equal(t, original.codec.Name(), got.codec.Name())
+	assert.Equal(t, original.treeName, got.treeName)
 }
 
-// TestUnmarshalJSON test the UnmarshalJSON function.
 func TestUnmarshalJSON(t *testing.T) {
-	treeName := "testing_unmarshalJSON_tree"
-	treeConstructorFn := sudoConstructorFn
-	err := RegisterTree(treeName, treeConstructorFn)
-	require.NoError(t, err)
+	t.Run("can unmarshal a custom tree", func(t *testing.T) {
+		// register tree
+		treeName := "test-unmarshal-json-tree"
+		err := RegisterTree(treeName, sudoConstructorFn)
+		require.NoError(t, err)
+		defer cleanUp(treeName)
 
-	codec := NewLeoRSCodec()
-	result, err := ComputeExtendedDataSquare([][]byte{
-		ones, twos,
-		threes, fours,
-	}, codec, treeConstructorFn)
-	if err != nil {
-		panic(err)
-	}
+		original, err := ComputeExtendedDataSquare([][]byte{ones, twos, threes, fours}, NewLeoRSCodec(), treeName)
+		require.NoError(t, err)
 
-	tests := []struct {
-		name             string
-		malleate         func()
-		expectedTreeName string
-		cleanUp          func()
-	}{
-		{
-			"Tree field exists",
-			func() {},
-			treeName,
-			func() {
-				cleanUp(treeName)
-			},
-		},
-		{
-			"Tree field missing",
-			func() {
-				// clear the tree name value in the eds before marshal
-				result.treeName = ""
-			},
-			DefaultTreeName,
-			func() {},
-		},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			test.malleate()
-			edsBytes, err := json.Marshal(result)
-			if err != nil {
-				t.Errorf("failed to marshal EDS: %v", err)
-			}
+		edsBytes, err := json.Marshal(original)
+		require.NoError(t, err)
 
-			var eds ExtendedDataSquare
-			err = json.Unmarshal(edsBytes, &eds)
-			if err != nil {
-				t.Errorf("failed to unmarshal EDS: %v", err)
-			}
-			if !reflect.DeepEqual(result.squareRow, eds.squareRow) {
-				t.Errorf("eds not equal after json marshal/unmarshal")
-			}
-			require.Equal(t, test.expectedTreeName, eds.treeName)
+		var got ExtendedDataSquare
+		err = got.UnmarshalJSON(edsBytes)
+		require.NoError(t, err)
 
-			test.cleanUp()
-		})
-	}
+		assert.Equal(t, original.dataSquare.Flattened(), got.dataSquare.Flattened())
+		assert.Equal(t, original.codec.Name(), got.codec.Name())
+		assert.Equal(t, original.treeName, got.treeName)
+	})
+	t.Run("unmarshals an EDS without a tree name using the default tree", func(t *testing.T) {
+		original, err := ComputeExtendedDataSquare([][]byte{ones, twos, threes, fours}, NewLeoRSCodec(), DefaultTreeName)
+		require.NoError(t, err)
+
+		original.treeName = ""
```

Comment:
[suggestion] Including a comment to explain the rationale behind leaving the tree name empty could enhance understanding, such as: "The tree name is intentionally left blank to test whether the Unmarshal process appropriately defaults to the default tree during unmarshaling"

Reply:
5c3532d36f3d0162555564b487a53f13b5484b6d